onc-healthit/lantern-back-end
# Creating an Archive File

This ticket generates an archive file in the format given by the attached archive file format, with adjustments made in Wes's email from 1/5/21.

[archival-file-documentation.xlsx](https://github.com/onc-healthit/lantern-back-end/files/5954054/archival-file-documentation.xlsx)

**To test:**
- Put data in the database in whatever way you see fit
- Run `make create_archive start=<start date> end=<end date> file=<name of JSON file>`, where the dates are in the format `2021-01-31`
- Make sure the data written to the test file specified is correct

--------------------------

Pull requests into this repository require the following checks to be completed by the submitter and reviewers.

**Submitter:**
  - This pull request describes why these changes were made.
  - The JIRA ticket links to this PR.
  - The JIRA ticket for this PR is: https://oncprojectracking.healthit.gov/support/browse/LANTERN-296
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.
  - Code diff has been done and checked to ensure only expected code is being committed.
  - A linter for the language has been run.

**Primary Reviewer:**

  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
  - Code is maintainable, reusable, efficient, and correct.
  - Code accomplishes the tasks purpose.
  - Code follows style guidance appropriate for the language, including passing linter checks.
  - Code is well commented (which does not mean verbosely commented).
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.

**Secondary Reviewer:**

  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
  - Code accomplishes the tasks purpose.
  - Tests have been run locally and pass.

**Merger:**

Prior to merging, ensure that all checks have been completed. Delete the branch after the merge
is complete.

Create a new branch `<branch name>_gomodupdate`. Run `make update_mods` to update the `go.mod`
and `go.sum` files to point to master. Create a PR. Require one sanity check reviewer.
